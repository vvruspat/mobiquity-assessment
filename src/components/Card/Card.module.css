.card {
    border-radius: 12px;
    background: var(--background-lighter-color);
}

.card.card_accent {
    background: var(--background-accent-color);
}

.header {
    padding: var(--padding-regular) var(--padding-large);
    color: var(--regular-color);
    background: var(--background-dark-color);
    border-top-left-radius: 12px;
    border-top-right-radius: 12px;
    text-transform: uppercase;
}

.card.card_accent .header {
    color: var(--regular-color);
    background: var(--accent-color);
}

.content {
    padding: var(--padding-large);
}