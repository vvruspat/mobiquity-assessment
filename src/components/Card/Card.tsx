import { FC, PropsWithChildren, ReactNode } from "react";
import cn from "clsx";

import s from "./Card.module.css";

type CardProps = {
  /** Make card highlighted */
  isAccent: boolean;
  /** Card header */
  header?: ReactNode;
};

/**
 * Card component
 */
export const Card: FC<PropsWithChildren<CardProps>> = ({
  isAccent = false,
  children,
  header,
}) => {
  return (
    <section
      className={cn(s.card, {
        [s.card_accent]: isAccent,
      })}
    >
      {header && <header className={s.header}>{header}</header>}
      <div className={s.content}>{children}</div>
    </section>
  );
};
