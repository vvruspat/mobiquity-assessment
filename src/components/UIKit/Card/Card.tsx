import {
  DetailedHTMLProps,
  FC,
  HTMLAttributes,
  PropsWithChildren,
  ReactNode,
} from "react";
import cn from "clsx";

import s from "./Card.module.css";

type CardProps = DetailedHTMLProps<
  HTMLAttributes<HTMLDivElement>,
  HTMLDivElement
> & {
  /** Make card highlighted */
  isAccent?: boolean;
  /** Card header */
  header?: ReactNode;
};

/**
 * Card component
 */
export const Card: FC<PropsWithChildren<CardProps>> = ({
  isAccent = false,
  children,
  header,
  className,
  ...htmlProps
}) => {
  return (
    <section
      className={cn(
        s.card,
        {
          [s.card_accent]: isAccent,
        },
        className
      )}
      {...htmlProps}
    >
      {header && <header className={s.header}>{header}</header>}
      <div className={s.content}>{children}</div>
    </section>
  );
};
