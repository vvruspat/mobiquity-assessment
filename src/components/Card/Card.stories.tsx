import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Card } from "./Card";
import "../../index.css";

export default {
  title: "UI Kit/Card",
  component: Card,
} as ComponentMeta<typeof Card>;

const Template: ComponentStory<typeof Card> = (args) => (
  <Card {...args}>Card content</Card>
);

export const CardTemplate = Template.bind({});

CardTemplate.args = {
  isAccent: false,
  header: "Winner 2022",
};
