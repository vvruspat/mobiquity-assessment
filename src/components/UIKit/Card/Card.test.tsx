import { render, screen } from "@testing-library/react";
import { Card } from "./Card";

test("renders card", () => {
  render(<Card>Test card</Card>);

  const el = screen.getByTestId("card");

  expect(el).toHaveTextContent("Test card");
  expect(el).toMatchSnapshot();
});

test("renders card accent", () => {
  render(<Card isAccent>Test card</Card>);

  const el = screen.getByTestId("card");

  expect(el).toHaveClass("card_accent");
  expect(el).toMatchSnapshot();
});

test("renders card header", () => {
  render(<Card header="Title">Test card</Card>);

  const el = screen.getByTestId("card");

  expect(el).toHaveTextContent("Title");
  expect(el).toHaveTextContent("Test card");
  expect(el).toMatchSnapshot();
});
