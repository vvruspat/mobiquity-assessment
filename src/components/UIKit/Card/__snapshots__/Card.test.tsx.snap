// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders card 1`] = `
<section
  class="card"
  data-testid="card"
>
  <div
    class="content"
  >
    Test card
  </div>
</section>
`;

exports[`renders card accent 1`] = `
<section
  class="card card_accent"
  data-testid="card"
>
  <div
    class="content"
  >
    Test card
  </div>
</section>
`;

exports[`renders card header 1`] = `
<section
  class="card"
  data-testid="card"
>
  <header
    class="header"
  >
    Title
  </header>
  <div
    class="content"
  >
    Test card
  </div>
</section>
`;
